mrmustard/physics/ansatze.py

```diff
@@ -1092,6 +1100,35 @@ def from_function(cls, fn: Callable, **kwargs: Any) -> ArrayAnsatz:
         ret._kwargs = kwargs
         return ret
 
+    def reduce(self, shape: int | Sequence[int]) -> ArrayAnsatz:
+        r"""
+        Returns a new ``ArrayAnsatz`` with a sliced array.
+
+        Args:
+            shape: The shape of the array of the returned ``ArrayAnsatz``.
+        """
+        if shape == self.array.shape[1:]:
+            return self
+        length = self.num_vars
+        shape = (shape,) * length if isinstance(shape, int) else shape
+        if len(shape) != length:
+            msg = f"Expected shape of length {length}, "
+            msg += f"given shape has length {len(shape)}."
+            raise ValueError(msg)
+
+        if any(s > t for s, t in zip(shape, self.array.shape[1:])):
+            warn(
```

Comment:
Do we still want this warning? Or should the recommendation just be included in the docstring

Reply:
perhaps it is a bit too annoying indeed